wazo_confd/plugins/user/middleware.py

```diff
@@ -1,22 +1,103 @@
+        self._middleware_handle = middleware_handle
+        self._schema = UserListItemSchema()
 
-    def create(self, body, tenant_uuid):
+    def create(self, body, tenant_uuid, tenant_uuids):
         form = self._schema.load(body)
         form['tenant_uuid'] = tenant_uuid
+
+        auth = form.pop('auth', None)
+        lines = form.pop('lines', None) or []
+        incalls = form.pop('incalls', None) or []
+        groups = form.pop('groups', None) or []
+        switchboards = form.pop('switchboards', None) or []
+
         model = User(**form)
         model = self._service.create(model)
-        return self._schema.dump(model)
+        user_dict = self._schema.dump(model)
+        user_dict['lines'] = []
+        user_dict['incalls'] = []
+        user_dict['groups'] = []
+        user_dict['switchboards'] = []
+
+        if auth:
+            auth['uuid'] = user_dict['uuid']
+            auth['tenant_uuid'] = user_dict['tenant_uuid']
+            user_dict['auth'] = self._wazo_user_service.create(auth)
+
+        if lines:
+            for line_body in lines:
+                line = self._middleware_handle.get('line').create(
+                    line_body, tenant_uuid, tenant_uuids
+                )
+                self._middleware_handle.get('user_line_association').associate(
+                    user_dict['id'], line['id'], tenant_uuids
+                )
+                user_dict['lines'].append(line)
+            Session.expire(model)
+
+        if incalls:
+            for incall_body in incalls:
+                incall = self._middleware_handle.get('incall').create(
+                    {'destination': {'type': 'user', 'user_id': user_dict['id']}},
+                    tenant_uuid,
+                )
+                incall_body['id'] = incall['id']
+                user_dict['incalls'].append(incall_body)
+
+                for extension in incall_body['extensions']:
+                    # create extension (the "source/origin/public/external")
+                    source_extension_body = {
+                        'context': extension['context'],
+                        'exten': extension['exten'],
+                    }
+                    source_extension = self._middleware_handle.get('extension').create(
+                        source_extension_body, tenant_uuids
+                    )
+                    extension['id'] = source_extension['id']
+
+                    # link incall+source extension
+                    self._middleware_handle.get(
+                        'incall_extension_association'
+                    ).associate(incall['id'], source_extension['id'], tenant_uuids)
+            Session.expire(model)
+
+        if groups:
+            self._middleware_handle.get('user_group_association').associate_all_groups(
+                {'groups': groups}, user_dict['uuid']
+            )
+            Session.expire(model)
+        user_dict['groups'] = groups
+
+        if switchboards:
```

Comment:
You don't need the `if` here since there's only a `for`. Otherwise you can remove the `or []` when popping switchboards